src/workerd/api/worker-rpc.c++

```diff
@@ -2,19 +2,175 @@
 // Licensed under the Apache 2.0 license found in the LICENSE file or at:
 //     https://opensource.org/licenses/Apache-2.0
 
-#pragma once
-
 #include <workerd/api/worker-rpc.h>
+#include <workerd/api/global-scope.h>
 
 namespace workerd::api {
 
 kj::Maybe<jsg::JsValue> WorkerRpc::getNamed(jsg::Lock& js, kj::StringPtr name) {
-    // TODO(soon): The server side PR will fill this function with the RPC call and other stuff,
-    //             this is just a temporary placeholder for testing.
-    return jsg::JsValue(js.wrapReturningFunction(js.v8Context(),
-        [](jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>& args) {
-          return js.error("WorkerRpc is unimplemented");
+  // Named intercept is enabled, this means we won't default to legacy behavior.
+  return jsg::JsValue(js.wrapReturningFunction(js.v8Context(), [this, methodName=kj::str(name)]
+      (jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>& args) {
```

Comment:
suggestion: I think you might be able to break up this big function to be easier to scan:
```
struct ResponseAndResult{
  capnp::Response<rpc::JsRpcTarget::CallResults> response;
  WorkerInterface::CustomEvent::Result result;
};
kj::Promise<ResponseAndResult> WorkerRpc::sendRpc(...) {
  ...
  auto result = co_await worker->customEvent(kj::mv(event));
  auto response = co_await builder.send());
  return ResultAndResponse{
    .result = result,
    .response = kj::mv(response),
  };
}
kj::Maybe<jsg::JsValue> WorkerRpc::getNamed(jsg::Lock& js, kj::StringPtr name) {
  return jsg::JsValue(js.wrapReturningFunction(js.v8Context(), [this, methodName=kj::str(name)]
        (jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>& args) {
    return js.wrapSimplePromise(ioContext.awaitIo(startRpc(...), [this](auto response){ return handleResponse(response); })
  };
}

Reply:
https://github.com/cloudflare/workerd/compare/5556c7bce286e193e5956e3e311b439d60956985..9f2a948b740bb6157c955458aed1f2a737722f72

Thanks for the suggestion, I added a few more comments and dropped an attach since we're `co_await`ing promises so it wasn't necessary.